src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null) {
+        result += `${separator}`;
+      } else if (typeof this[i] === 'undefined' && i === this.length - 1) {
+        continue;
+      } else if (typeof this[i] === 'undefined') {
+        result += `${separator}`;
+      } else if (i === this.length - 1) {
+        result += `${this[i]}`;
+      } else {
+        result += `${this[i]}${separator}`;
+      }
```

Comment:
I think 2 if-blocks should be enough
one to check whether the element is undefined or null and second to check the last element